weblab_cg/tests/test_chaste_cg_develop.py

```diff
@@ -367,6 +366,30 @@ def _resolve_linkers(self, equation_list, subs_dict):
 
         return equation_list, subs_dict
 
+    def _resolve_remaining_linkers(self, expected, generated):
+        differing_expected = self._different_eqs(generated, expected)[1]
+        for dif_exp in differing_expected:
+            delete = False
+            for j in range(len(expected)):
+                if dif_exp != expected[j]:
+                    old = expected[j][1]
+                    expected[j][1] = expected[j][1].subs({self._get_var_name(dif_exp[0]):
+                                                          dif_exp[1]})
+                    delete = delete or old != expected[j][1]
+                    for k in range(len(generated)):
+                        if expected[j][0] == generated[k][0]:
+                            generated[k][1] = generated[k][1].subs({self._get_var_name(dif_exp[0]):
+                                                                    dif_exp[1]})
+            if delete:
+                expected_element_index = expected.index(dif_exp)
+                del expected[expected_element_index]
+        return expected, generated
+
+    def _different_eqs(self, expected, generated):
```

Comment:
You only ever read one of the returned pair, so I wouldn't make this calculate both. Instead, call the args say `main_eqs` and `filter_eqs` and calculate just
```
filter_lhss = [eq[0] for eq in filter_eqs]
return [eq for eq in main_eqs if eq[0] not in filter_lhss]
```

Reply:
ok